pkgs/applications/networking/instant-messengers/mikutter/default.nix

```diff
@@ -1,82 +1,109 @@
+{ lib
+, stdenv
+, fetchurl
+, bundlerEnv
+, alsaUtils
+, atk
+, bundix
+, bundler
+, common-updater-scripts
+, curl
+, desktop-file-utils
+, gobject-introspection
+, gtk2
+, jq
+, ruby
+, libnotify
+, which
+, wrapGAppsHook
+, writers
 }:
 
-# how to update:
-# find latest version at: http://mikutter.hachune.net/download#download
-# run these commands:
-#
-# wget http://mikutter.hachune.net/bin/mikutter.4.0.0.tar.gz
-# mkdir mikutter
-# cd mikutter
-# tar xvf ../mikutter.4.0.0.tar.gz
-# find . -not -name Gemfile -exec rm {} \;
-# find . -type d -exec rmdir -p --ignore-fail-on-non-empty {} \;
-# cd ..
-# mv mikutter/* .
-# rm mikutter.4.0.0.tar.gz
-# rm gemset.nix Gemfile.lock; nix-shell -p bundler bundix --run 'bundle lock && bundix'
+let
+  version = "4.1.4";
 
+  gems = bundlerEnv {
+    name = "mikutter-${version}-gems";
+    gemdir = ./deps;
+    groups = [ "default" "plugin" ];
+    inherit ruby;
+
+    # Avoid the following error:
+    # > `<module:Moneta>': uninitialized constant Moneta::Builder (NameError)
+    #
+    # Related:
+    # https://github.com/NixOS/nixpkgs/pull/76510
+    # https://github.com/NixOS/nixpkgs/pull/76765
+    # https://github.com/NixOS/nixpkgs/issues/83442
+    # https://github.com/NixOS/nixpkgs/issues/106545
+    copyGemFiles = true;
+  };
+
+  inherit (gems) wrappedRuby;
+in
 stdenv.mkDerivation rec {
   pname = "mikutter";
-  version = "4.0.0";
+  inherit version;
 
   src = fetchurl {
-    url = "https://mikutter.hachune.net/bin/mikutter.${version}.tar.gz";
-    sha256 = "0nx14vlp7p69m2vw0s6kbiyymsfq0r2jd4nm0v5c4xb9avkpgc8g";
+    url = "https://mikutter.hachune.net/bin/mikutter-${version}.tar.gz";
+    sha256 = "05253nz4i1lmnq6czj48qdab2ny4vx2mznj6nsn2l1m2z6zqkwk3";
   };
 
-  buildInputs = [ alsaUtils libnotify which gtk2 ruby atk gobject-introspection ];
-  nativeBuildInputs = [ wrapGAppsHook ];
+  nativeBuildInputs = [ desktop-file-utils wrapGAppsHook ];
+  buildInputs = lib.optionals stdenv.isLinux [
+    atk
+    gtk2
+    gobject-introspection
+    libnotify
+    which # some plugins use it at runtime
+    wrappedRuby
+  ] ++ lib.optionals stdenv.isLinux [ alsaUtils ];
 
-  unpackPhase = ''
-    mkdir source
-    cd source
-    unpackFile $src
-    rm -rf vendor
-  '';
+  postUnpack = "rm -rf vendor";
 
-  installPhase = let
-    env = bundlerEnv {
-      name = "mikutter-${version}-gems";
-      gemdir = ./.;
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/share/applications $out/opt/mikutter
 
-      inherit ruby;
-    };
-  in ''
     install -v -D -m644 README $out/share/doc/mikutter/README
     install -v -D -m644 LICENSE $out/share/doc/mikutter/LICENSE
-    rm -v README LICENSE
+    desktop-file-install --dir $out/share/applications \
+      --set-key Exec --set-value $out/bin/mikutter \
+      --set-key Icon --set-value $out/opt/mikutter/core/skin/data/icon.png \
+      deployment/appimage/mikutter.desktop
```

Comment:
Can we use makeDesktopItems here instead?

Reply:
Switched to `makeDesktopItem`.